interpreter/cling/include/cling/Interpreter/RuntimePrintValue.h

```diff
@@ -0,0 +1,162 @@
+//--------------------------------------------------------------------*- C++ -*-
+// CLING - the C++ LLVM-based InterpreterG :)
+// author:  Boris Perovic <boris.perovic@cern.ch>
+//
+// This file is dual-licensed: you can choose to license it under the University
+// of Illinois Open Source License or the GNU Lesser General Public License. See
+// LICENSE.TXT for details.
+//------------------------------------------------------------------------------
+#ifndef CLING_RUNTIME_PRINT_VALUE_H
+#define CLING_RUNTIME_PRINT_VALUE_H
+
+#include <string>
+#include "Value.h"
+
+#ifndef _Bool
+#define _Bool bool
+#endif
+
+namespace cling {
+
+  // void pointer
+  std::string printValue(void *ptr);
+
+  // Bool
+  std::string printValue(bool val);
+
+  // Chars
+  std::string printValue(char val);
+
+  std::string printValue(signed char val);
+
+  std::string printValue(unsigned char val);
+
+  // Ints
+  std::string printValue(short val);
+
+  std::string printValue(unsigned short val);
+
+  std::string printValue(int val);
+
+  std::string printValue(unsigned int val);
+
+  std::string printValue(long val);
+
+  std::string printValue(unsigned long val);
+
+  std::string printValue(long long val);
+
+  std::string printValue(unsigned long long val);
+
+  // Reals
+  std::string printValue(float val);
+
+  std::string printValue(double val);
+
+  std::string printValue(long double val);
+
+  // Char pointers
+  std::string printValue(const char *const val);
+
+  std::string printValue(char *val);
+
+  // std::string
+  std::string printValue(const std::string &val);
+
+  // cling::Value
+  std::string printValue(const Value &value);
+
+  // Maps declaration
+  template<typename CollectionType>
```

Comment:
Moved. Should I unify the namespaces? Currently printValue_New and printType_new (entry functions from Value::print) are in the namespace cling::valuePrinterInternal, whereas this would be cling::internal.